### ISSUE TITLE
XML error from Wunderlist workflow (Wunderlist API is down)

### ISSUE THREAD

**Issue by william-gill** [idpaterson/alfred-wunderlist-workflow#136]
Hi,
I'm getting an XML error back from the service when trying to add a new task via the workflow.
See [here](https://www.dropbox.com/s/jiya2uas281rxob/Screenshot%202016-09-21%2010.23.06.png?dl=0).

It was working previously, and stopped working this morning.
cheers

**Comment by csteinlehner** [comment]
I get this also sometimes (randomly at it seems). Most of the times if I try again, it works. Of course a little bit annoying, because if I don't see that there is this error, the todo is lost.

**Comment by aetjansen** [comment]
Same here. I get an XML error notification when I enter a task using the workflow. It worked fine until this morning indeed.

More specifically, this is what I get in the notification:
<?xml version="1.0" encoding="utf-8"?>...

Just an xml header, I know, but maybe it helps locating the source of the issue? A change on Wunderlist side of things?

**Comment by idpaterson** [comment]
This indeed seems to be an API problem. Looking at the debugger (beetle icon in Alfred's page for the workflow) I see the following:
```python
[2016-09-21 06:23:43][ERROR: action.script] 2016-09-21 06:23:43,294 workflow     ERROR    No JSON object could be decoded
Traceback (most recent call last):
  File "/Users/ian/Projects/wunderlist-swift/dist/workflow/workflow/workflow.py", line 2178, in run
    func(self)
  File "alfred-wunderlist-workflow.py", line 16, in main
    route(wf.args)
  File "/Users/ian/Projects/wunderlist-swift/dist/workflow/wunderlist/handlers/route.py", line 77, in route
    handler.commit(command, modifier)
  File "/Users/ian/Projects/wunderlist-swift/dist/workflow/wunderlist/handlers/new_task.py", line 163, in commit
    completed=task.completed)
  File "/Users/ian/Projects/wunderlist-swift/dist/workflow/wunderlist/api/tasks.py", line 82, in create_task
    info = req.json()
  File "/Users/ian/Projects/wunderlist-swift/dist/workflow/requests/models.py", line 826, in json
    return complexjson.loads(self.text, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by aetjansen** [comment]
If it helps, I am not on Sierra yet. To me it occurred while running OSX 10.11.6, using the latest version of your workflow on Alfred v3.1.1 [737] and Wunderlist 3.4.4 (40).

**Comment by idpaterson** [comment]
> HTTP/1.1 503 Service Unavailable: Back-end server is at capacity

This will continue to be a problem until Wunderlist gets it under control. Meanwhile it is a good opportunity to work on making the workflow fail more informatively.

For anyone experiencing this problem, trying again later may work. Alfred makes that a bit easier by storing recent queries – in case you don't already using this feature the up arrow can be used to recall past task entries:
![past queries](https://cloud.githubusercontent.com/assets/507058/18707908/370a6224-7fc6-11e6-8873-1d45e429b39f.png)

**Comment by idpaterson** [comment]
I'll have to patch alfred-workflow in order to get the error to show as text rather than that XML blob. Thanks everyone for reporting and checking back in!

**Comment by aetjansen** [comment]
Thank you for your quick response. A more meaningful error would be handy, maybe include a reference to Wunderlist's status url?

**Comment by idpaterson** [comment]
I like the idea of linking to the Wunderlist status page. Alfred will not allow me to do anything fancy when you actually submit the task (only a few lines of text for the notification) but I could trigger Alfred to show up again with more information about the error and a link to the status page. I'll give that some thought. It is infrequent enough that I don't want to put too much effort into it but like @csteinlehner mentioned it is currently easy to not notice the error and lose that task.

**Comment by idpaterson** [comment]
Well, I did not anticipate having multiple days of downtime otherwise probably would have pushed out a quick fix to avoid showing the error for others. I may consider working on the idea proposed by @aetjansen this evening (it is morning now) if Wunderlist is still down.

**Comment by idpaterson** [comment]
Wunderlist is back online. I will close this issue tomorrow if it looks like the team has permanently resolved the API issues and open a new one for the improved error handling work.

**Comment by william-gill** [comment]
Thanks for working on this and keeping us up to date!
I can also confirm the workflow is now working again as expected. :)

**Comment by idpaterson** [comment]
#139 will track the improved error handling. For 0.6.2 I patched alfred-workflow to show the correct text for errors rather than XML code.

**Comment by csteinlehner** [comment]
After updating to 0.6.2 I always get the XML error as notification, but the Task gets created.

**Comment by idpaterson** [comment]
Can you see what the full error message is in the workflow's debugging screen in Alfred?

**Comment by csteinlehner** [comment]
unfortunately there is nothing in the debug log. I just have this notification 
![bildschirmfoto 2016-09-26 um 14 17 08](https://cloud.githubusercontent.com/assets/275506/18832430/05b80622-83f4-11e6-989b-86c2c9f8417f.png)

**Comment by idpaterson** [comment]
Got it, thanks. A few people have downloaded 0.6.2 already so I'm going to push out the fix as 0.6.3 =\

**Comment by idpaterson** [comment]
[0.6.3](https://github.com/idpaterson/alfred-wunderlist-workflow/releases/tag/0.6.3) is up
